### ISSUE TITLE
Datastore Model with custom ownerField fails

### ISSUE THREAD

**Issue by guustmc** [aws-amplify/amplify-swift#3026]
### Describe the bug

CLI Version: 12.1.0
SDK version: 2.11.5

When using the following Datastore model on Amplify studio & running amplify pull:

```
type UserChild @model @auth(rules: [{allow: owner, ownerField: "authors"}]) {
  id: ID!
  name: String
  gender: UserChildGender
  userID: ID! @index(name: "byUser")
  dueDate: AWSDate
  authors: [String]
}
```

And then calling:
```
let child = Child(name: "test", gender: .boy, userID: "xyz", dueDate: Date())
try await Amplify.DataStore.save(child)
```

I get the following error:
```
GraphQLResponseError<MutationSync<AnyModel>>: 
GraphQL service returned a successful response containing errors: [
      Amplify.GraphQLError(
          message: "Validation error of type UnknownArgument: Unknown field argument authors @ 'onUpdateUserChild\", 
          locations: nil, path: nil, extensions: nil
      )]
```

This error only happens when im using the custom ownerField 'authors' (I need to share user data between multiple users).  If I use the default owner code everything works nicely with authentication.

I've tried removing all local amplify data & pulling the project again.

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Create an amplify project
2. Use the data model supplied in the report
3. Pull the project using the amplify CLI (without any changes to the config)
4. Authenticate yourself using the Amplify auth plugin & cognito 
4. Save a record of the model to datastore
```

### Expected behavior

The model is saved with the saving user added as one of the authors. 
The data should be:
authors: ['xyz::xyz']

### Amplify Framework Version

iOS SDK version 2.11.5

### Amplify Categories

Auth, DataStore

### Dependency manager

Swift PM

### Swift version

5

### CLI version

12.1.0

### Xcode version

14.3.1

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>

[RetryableGraphQLSubscriptionOperation<MutationSync<AnyModel>>] APIError: Subscription item event failed with error
Caused by:
GraphQLResponseError<MutationSync<AnyModel>>: GraphQL service returned a successful response containing errors: [Amplify.GraphQLError(message: "Validation error of type UnknownArgument: Unknown field argument authors @ \'onDeleteUserChild\'", locations: nil, path: nil, extensions: nil)]
Recovery suggestion: The list of `GraphQLError` contains service-specific messages
2023-06-16 15:15:55.878464+0200 [37392:4128790] [RetryableGraphQLSubscriptionOperation<MutationSync<AnyModel>>] [37E8B0FD-6D46-4869-87BC-544CBCD2DE61] - Failed
2023-06-16 15:15:55.878490+0200 [37392:4128790] [IncomingAsyncSubscriptionEventPublisher] [InitializeSubscription.1] API.subscribe failed for `UserChild` error: Subscription item event failed with error
2023-06-16 15:15:55.878490+0200 [37392:4128801] [RetryableGraphQLSubscrip2023-06-16 15:15:55.878490+0200 [37392:4128805] [RetryableGraphQLSubscriptionOperation<MutationSync<AnyModel>>] APIError: Subscription item event failed with error
Caused by:
GraphQLResponseError<MutationSync<AnyModel>>: GraphQL service returned a successful response containing errors: [Amplify.GraphQLError(message: "Validation error of type UnknownArgument: Unknown field argument authors @ \'onUpdateUserChild\'", locations: nil, path: nil, extensions: nil)]
Recovery suggestion: The list of `GraphQLError` contains service-specific messages
</details>
```

### Is this a regression?

No

### Device

iPhone 14

### iOS Version

16

### Specific to simulators

No

**Comment by pixelier** [comment]
There is a [draft PR](https://github.com/aws-amplify/amplify-swift/pull/1606) dating from January 2022. 
Will it be fixed anytime soon? 🙏

**Comment by atierian** [comment]
Thanks for reporting this. We'll investigate and respond here with any updates.

**Comment by lawmicha** [comment]
This was released in https://github.com/aws-amplify/amplify-swift/releases/tag/2.19.0

**Comment by lawmicha** [comment]
Closing since this has been released, please try out 2.19.0. Feel free to let us know to re-open or open a new issue with your schema if something isn't working as expected
